[INFO -- 2018-11-05 21:15:57]
Starting flip_value script
version: 1.1
[INFO -- 2018-11-05 21:15:57]
Initialization strings:
[INFO -- 2018-11-05 21:15:57]
(1) file /tmp/lreg/lreg
[INFO -- 2018-11-05 21:15:57]
(2) set args /tmp/lreg/randInput /tmp/lreg/output /tmp/lreg/detected.log
[INFO -- 2018-11-05 21:15:57]
initSignal:0
[INFO -- 2018-11-05 21:15:57]
endSignal:10
[INFO -- 2018-11-05 21:15:57]
faultModel:0
[EXCEPTION -- 2018-11-05 21:15:59]
pythonException: empty range for randrange() (0,0, 0)
[EXCEPTION -- 2018-11-05 21:16:00]
pythonException: empty range for randrange() (0,0, 0)
[EXCEPTION -- 2018-11-05 21:16:03]
pythonException: Unable to exit loop in struct fields; Exiting wihtout making a bit flip
[INFO -- 2018-11-05 21:16:05]
Backtrace BEGIN:
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
#1  0x00007ffff6ec6801 in __GI_abort () at abort.c:79
#2  0x00007ffff6f0f897 in __libc_message (action=action@entry=do_abort, fmt=fmt@entry=0x7ffff703cb9a "%s\n") at ../sysdeps/posix/libc_fatal.c:181
#3  0x00007ffff6f1690a in malloc_printerr (str=str@entry=0x7ffff703ad88 "free(): invalid pointer") at malloc.c:5350
#4  0x00007ffff6f1de1c in _int_free (have_lock=0, p=0x555555772920, av=0x7ffff7271c40 <main_arena>) at malloc.c:4157
#5  __GI___libc_free (mem=0x555555772930) at malloc.c:3124
#6  0x0000555555557cc2 in __gnu_cxx::new_allocator<double>::deallocate (this=0x55555575b160 <y>, __p=0x555555772930) at /usr/include/c++/7/ext/new_allocator.h:125
#7  0x00005555555574d1 in std::allocator_traits<std::allocator<double> >::deallocate (__a=..., __p=0x555555772930, __n=72) at /usr/include/c++/7/bits/alloc_traits.h:462
#8  0x0000555555556b7c in std::_Vector_base<double, std::allocator<double> >::_M_deallocate (this=0x55555575b160 <y>, __p=0x555555772930, __n=72) at /usr/include/c++/7/bits/stl_vector.h:180
#9  0x0000555555556839 in std::_Vector_base<double, std::allocator<double> >::~_Vector_base (this=0x55555575b160 <y>, __in_chrg=<optimized out>) at /usr/include/c++/7/bits/stl_vector.h:162
#10 0x0000555555556225 in std::vector<double, std::allocator<double> >::~vector (this=0x55555575b160 <y>, __in_chrg=<optimized out>) at /usr/include/c++/7/bits/stl_vector.h:435
#11 0x00007ffff6ec9041 in __run_exit_handlers (status=0, listp=0x7ffff7271718 <__exit_funcs>, run_list_atexit=run_list_atexit@entry=true, run_dtors=run_dtors@entry=true) at exit.c:108
#12 0x00007ffff6ec913a in __GI_exit (status=<optimized out>) at exit.c:139
#13 0x00007ffff6ea7b9e in __libc_start_main (main=0x555555555b15 <main(int, char**)>, argc=4, argv=0x7fffffffde48, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7fffffffde38) at ../csu/libc-start.c:344
#14 0x000055555555501a in _start ()

46	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.

Backtrace END
Fliping value of field: priv
Fliping value of field name: data
Fliping value of field: prev
Fliping a bit of the pointer
Memory content before bitflip:0x000x000x000x000x000x000x000x00
Fault Model: Single bit-flip
base address to flip value: 0x7fffffffddd8
address max offset: 8
Memory content after  bitflip:0x000x000x000x080x000x000x000x00
frame name: __libc_start_main
symbol name: unwind_buf
symbol filename: ../csu/libc-start.c
symbol line: 294
value: {cancel_jmp_buf = {{jmp_buf = {0, 5972110868876211867, 93824992235504, 140737488346688, 0, 0, 555181238826940059, 555196534078289563}, mask_was_saved = 0}}, priv = {pad = {0x8000000, 0x0, 0x7ffff7de5733 <_dl_init+259>, 0x7ffff7dc52b8}, data = {prev = 0x8000000, cleanup = 0x0, canceltype = -136423629}}}
value address: 0x7fffffffdd90
Type: The type is a structure.
Type sizeof: 104
Field name: cancel_jmp_buf
Field Type: The type is an array.
Field Type sizeof: 72
Field name: priv
Field Type: The type is a union.
Field Type sizeof: 32

[INFO -- 2018-11-05 21:16:05]
Fault Injection Successful after 8.385374069213867s
[INFO -- 2018-11-05 21:16:05]
event type: exit
[EXCEPTION -- 2018-11-05 21:16:05]
exit code: no exit code available
